File: 2.server/src/main/java/ru/vitaliy/petrov/server/forms/requests/documents/AdministrativeOffenceSceneSchemeAddRequest.java
package ru.vitaliy.petrov.server.forms.requests.documents;

import lombok.Data;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;
import java.sql.Date;
import java.sql.Time;

@Data
public class AdministrativeOffenceSceneSchemeAddRequest {

    @NotNull
    private final Date dateOfFill;

    @NotNull
    private final Time timeOfFill;

    @NotNull
    @NotEmpty
    private final String placeOfFill;

    @NotNull
    private final Long policeOfficerID;

    @NotNull
    @NotEmpty
    private final String fileLink;

    @NotNull
    @NotEmpty
    private final String schemeConventions;

    @NotNull
    private final Long firstWitnessID;

    @NotNull
    private final Long secondWitnessID;

    @NotNull
    private final Long carAccidentEntityID;
}

File: client/app/src/main/java/com/petrov/vitaliy/caraccidentapp/domain/models/requests/documents/AdministrativeOffenceSceneSchemeAddRequest.kt
package com.petrov.vitaliy.caraccidentapp.domain.models.requests.documents

import java.sql.Date
import java.sql.Time

data class AdministrativeOffenceSceneSchemeAddRequest(
    val dateOfFill: Date,
    val timeOfFill: Time,
    val placeOfFill: String,
    val policeOfficerID: Long,
    val fileLink: String,
    val schemeConventions: String,
    val firstWitnessID: Long,
    val secondWitnessID: Long,
    val carAccidentEntityID: Long
)
